### ISSUE TITLE
Documentation error: `save` argument in Python API does not work when passed to predictor()

### ISSUE THREAD

**Issue by dingjie-ai** [ultralytics/ultralytics#22977]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

# Documentation issue: `save` argument in Python API does not work when passed to predictor()

## 🐛 Issue

The Python API documentation for `SAM3SemanticPredictor` shows examples like this:

```python
from ultralytics.models.sam import SAM3SemanticPredictor

overrides = dict(
    conf=0.25,
    task="segment",
    mode="predict",
    model="sam3.pt",
    half=True,
)

predictor = SAM3SemanticPredictor(overrides=overrides)
results = predictor(text=["a person"], save=True)
It suggests that passing save=True in the function call will save annotated images.

##Problem
In practice, save=True passed this way is only received in kwargs inside stream_inference, and the check:
if self.args.save or self.args.save_txt,will not be triggered. As a result, files are always saved or the argument is ignored, depending on context. This makes the documentation misleading.

##Suggested Fix
Move the save argument into overrides when initializing the predictor:
```python
overrides = dict(
    conf=0.25,
    task="segment",
    mode="predict",
    model="sam3.pt",
    half=True,
    save=True,  # ✅ now save works
)

predictor = SAM3SemanticPredictor(overrides=overrides)
results = predictor(text=["a person"])

This ensures that save behaves as documented and is correctly recognized by the internal args logic.

### Environment

OS                  Windows-10-10.0.26100-SP0
Environment         Windows
Python              3.8.18
Install             pip
RAM                 23.82 GB
Disk                188.6/376.4 GB
CPU                 12th Gen Intel Core(TM) i5-12400F
CPU count           12
GPU                 NVIDIA GeForce RTX 2060, 12288MiB
GPU count           1
CUDA                11.8

numpy               ✅ 1.23.4>=1.23.0
numpy               ✅ 1.23.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.7.4>=3.3.0
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 10.1.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 2.4.1+cu118>=1.8.0
torch               ✅ 2.4.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.19.1+cu118>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.6
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0





### Minimal Reproducible Example

from ultralytics.models.sam import SAM3SemanticPredictor

# ⚡ Minimal Reproducible Example (MRE)

# 1️⃣ Predictor configuration
overrides = dict(
    conf=0.25,
    task="segment",
    mode="predict",
    model="sam3.pt",
    half=True,
    save=True,  # ✅ save must be in overrides
)

predictor = SAM3SemanticPredictor(overrides=overrides)

# 2️⃣ Set the input image
predictor.set_image("path/to/image.jpg")

# 3️⃣ Run inference
# Do NOT pass save=True here; it will be ignored
results = predictor(text=["person", "bus"])

# 4️⃣ Print results
for r in results:
    print("Boxes:", r.boxes)
    print("Masks:", r.masks)


### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @dingjie-ai, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon ✅.

If this is a 🐛 Bug Report (or docs/API mismatch), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. You included an MRE already (thanks!)—if you can, please also add the *exact* expected vs. actual behavior (e.g., whether files are saved when `save=True` is passed at call-time, and where outputs are written), plus any console logs.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
@dingjie-ai Can you open a PR?

**Comment by dingjie-ai** [comment]
> [@dingjie-ai](https://github.com/dingjie-ai) Can you open a PR?

Yes. I’ll open a docs PR to fix the documentation and add the workaround.

**Comment by dingjie-ai** [comment]
> [@dingjie-ai](https://github.com/dingjie-ai) Can you open a PR?

@Y-T-G This PR addresses the documentation issue in #22977
